src/Messaging/src/RabbitMQ/Attributes/DeclareQueueBaseAttribute.cs

```diff
@@ -4,6 +4,7 @@
 
 namespace Steeltoe.Messaging.RabbitMQ.Attributes;
 
+[AttributeUsage(AttributeTargets.Class)]
 public abstract class DeclareQueueBaseAttribute : Attribute
```

Comment:
All derived types are annotated with:
```c#
[AttributeUsage(AttributeTargets.Class | AttributeTargets.Interface | AttributeTargets.Method, AllowMultiple = true)]
```
Is this difference intentional?

Reply:
no, I missed that. I'll add those too

Reply:
Actually looking at this again... Given that it is a BaseAttribute I would think that it should only be used on classes

Reply:
why?

Reply:
I'll start by saying I'm not an expert on how this is supposed to be used, but looking at how we're currently using it, it's only in classes. Given that it's `abstract` it can't be used directly, so how or why would you use it any other way?

Reply:
This is explained at https://docs.microsoft.com/en-us/dotnet/api/system.attributeusageattribute.inherited?view=net-6.0.
Because `ValidOn` defaults to `All` (which is always wrong in our case), the best we can do is set it to `Class` because there is no `None`. But more importantly, we should set `Inherited` to `false`, which basically ensures that the `AttributeUsage` here is ignored by derived classes.